### ISSUE TITLE
[bug]: Manage Flow Instance Variable throws ORA-6502 when page item is  is currency (with $ sign)

### ISSUE THREAD

**Issue by ujnak** [flowsforapex/apex-flowsforapex#406]
### Flows for APEX version

21.1

### Oracle Database version

18

### Oracle APEX version

21.2

### What happened?

Plugin **Manage Flow Instance Variable**  throws ORA-6502 when NUMBER field has format mask to represent Currency i.e. FML999G999G999G999G990D00.

In function execution of flow_plugin_manage_instance_variables.pkb,

Line: 482 - 488
```
                 case l_attribute4
                     when 'set' then
                        flow_process_vars.set_var(
                           pi_prcs_id    => l_prcs_id
                           , pi_var_name   => l_prcs_var_name
                           , pi_num_value  => to_number( apex_util.get_session_state( p_item => l_item_name ) )
                        );
```
It seems that format mask is required to convert currency to number, i.e.
```
case
when l_items(item_name).format_mask is not null then
   to_number( apex_util.get_session_state( p_item => l_item_name ), l_items(item_name).format_mask )
else
   to_number( apex_util.get_session_state( p_item => l_item_name ) )
end
```
Line 413 - 416
Initialization code for l_items also need to be updated, i.e.
```
if rec.item_type = 'NUMBER' and rec.format_mask is not null then
    l_items(rec.item_name).format_mask := rec.format_mask;
elsif rec.item_type = 'DATE' then
    l_items(rec.item_name).format_mask := coalesce( rec.format_mask, v('APP_NLS_DATE_FORMAT') );
end if;
```

**Comment by LouisMoreaux** [comment]
Hello, 

Thanks for letting us know about this issue. It seems to be linked to a change behaviour in APEX 21.1 referenced here: https://docs.oracle.com/en/database/oracle/application-express/21.1/htmrn/#GUID-3AAF431E-D7AA-46F8-A03A-15F03B6333F8

We will investigate and provide a fix as soon as possible.

**Comment by rallen2010** [comment]
Fixed in 22.1.   Closing
